File: JAVA/HDUOJ2045/src/Main.java
import java.util.Scanner;
public class Main {
	public static void main(String [] args) {
		Scanner input = new Scanner(System.in);
		int a;
		int arr[] = new int[52];
		arr[0]=3;
		arr[1]=6;
		arr[2]=6;
		for(int i=3;i<arr.length;i++) {
			arr[i] = arr[i-1]+arr[i-2]*2;
		}
		while(input.hasNext()) {
			a = input.nextInt();
			System.out.println(arr[a-1]);
		}
	} 
}

File: C/HDUOJ2045/Main.cpp
#include<iostream>
using namespace std;
int main()
{
    long long arr[52];
    arr[1]=3;
    arr[2]=6;
    arr[3]=6;
    for(int i=4;i<=50;i++)
    {
        arr[i]=arr[i-1]+2*arr[i-2];
    }
    int n;
    while(cin>>n)
    {
        cout<<arr[n]<<endl;
    }
    return 0;
}
